p2p/pex/pex_reactor.go

```diff
@@ -504,7 +504,7 @@ func (r *Reactor) ensurePeers() {
 	if r.book.NeedMoreAddrs() {
 		// 1) Pick a random peer and ask for more.
 		peer := r.Switch.Peers().Random()
-		if peer != nil {
+		if peer != nil && ensurePeersPeriodElapsed {
```

Comment:
Two things ,are we not still dialing peers in L487? If that is also not dialed when `ensurePeersPeriodElapsed` is `false`, are we not effectively back to the same behaviour before the fix in #3360 ? 

I might be missing something , but let's say the value of `ensurePeersPeriod`  is 30s. We got peer addresses at 

`t := 10s` . Should we not only wait for 
`tWait := ensurePeersPeriod - t` ?  

We can't have it both ways, dial immediately and make sure the period is respected. Maybe I am missing something here.

Reply:
Ok for 1. I realized that that one is not a `PexRequest` so it won't cause any problems. 

And for 2 I am also sure now that it is fine because we actually want the action above in 487 to happen but we do not want to issue a `RequestPeers` request.

Reply:
I also thought about 2, but came to the conclusion that it's still useful because we would be trying to pick peers from the address book and dial them; we just won't be sending `PexRequests` to them if we need more addresses.